ui/app/services/capabilities.ts

```diff
@@ -0,0 +1,91 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: BUSL-1.1
+ */
+
+import Service, { service } from '@ember/service';
+import { assert } from '@ember/debug';
+
+import type AdapterError from '@ember-data/adapter/error';
+import type ArrayProxy from '@ember/array/proxy';
+import type CapabilitiesModel from 'vault/vault/models/capabilities';
+import type StoreService from 'vault/services/store';
+
+interface Query {
+  paths?: string[];
+  path?: string;
+}
+
+interface ComputedCapabilities {
+  canSudo: string;
+  canRead: string;
+  canCreate: string;
+  canUpdate: string;
+  canDelete: string;
+  canList: string;
+}
+export default class CapabilitiesService extends Service {
+  @service declare readonly store: StoreService;
+
+  request = async (query: Query) => {
+    if (query?.paths) {
+      const { paths } = query;
+      return this.store.query('capabilities', { paths });
+    }
+    if (query?.path) {
+      const { path } = query;
+      const storeData = await this.store.peekRecord('capabilities', path);
+      return storeData ? storeData : this.store.findRecord('capabilities', path);
+    }
+    return assert('query object must contain "paths" or "path" key', false);
+  };
+
+  /*
+  this method returns all of the capabilities for a particular path or paths
+  */
+  async fetchAll(query: Query): Promise<CapabilitiesModel | ArrayProxy<CapabilitiesModel>> | AdapterError {
```

Comment:
nit: pretty sure ember-data no longer uses `ArrayProxy` so it should be a regular array
```suggestion
  async fetchAll(query: Query): Promise<CapabilitiesModel | Array<CapabilitiesModel>> | AdapterError {
```

Reply:
I'm also wondering if we want to split out methods for multiple vs single paths -- using this gets complicated when we can expect either an array or single model

Reply:
I grabbed this type from what was being returned, but I suppose we haven't upgraded yet? So it's still coming in as a proxy array 🤔  I can update it